zipline/algorithm.py

```diff
@@ -204,11 +204,11 @@ def __init__(self, *args, **kwargs):
         # Update the TradingEnvironment with the provided asset metadata
         self.trading_environment = kwargs.pop('env',
                                               TradingEnvironment.instance())
-        self.trading_environment.update_asset_finder(
-            asset_finder=kwargs.pop('asset_finder', None),
-            asset_metadata=kwargs.pop('asset_metadata', None),
-            identifiers=kwargs.pop('identifiers', None)
+        self.trading_environment.write_data(
```

Comment:
Why is the pattern that we write data to the environment? I am not sure why we need to store this as global state.